### PR TITLE
Update supported native targets

### PR DESCRIPTION
**Pull request by qurbonzoda**

### CODE REVIEW

File: core/build.gradle.kts [hunk 1]
```diff
@@ -73,40 +81,28 @@ kotlin {
     sourceSets {
         commonMain {
             dependencies {
-                api("org.jetbrains.kotlin:kotlin-stdlib-common")
+                api("org.jetbrains.kotlin:kotlin-stdlib")
```
Comment (ilya-g) [inline core/build.gradle.kts:84]: This is not correct until stdlib becomes a mpp library. Better not to use stdlib dependency at all and let KGP configure it automatically.

Comment (qurbonzoda) [reply core/build.gradle.kts]: done

**Comment by Nailik** [comment]
Couldn't you bump kotlin to 1.9 already?
would remove the need of another PR.

**Comment by Nailik** [comment]
@ilya-g bump pls review asap that we finally have all targets and this can be used in multiplatform libraries

File: README.md [hunk 2]
```diff
@@ -9,7 +9,8 @@
 
 Immutable collection interfaces and implementation prototypes for Kotlin.
 
-This is a multiplatform library providing implementations for `jvm`, `js` (both Legacy and IR), `mingwX64`, `linuxX64` and Apple Kotlin/Native targets.
+This is a multiplatform library providing implementations for `jvm`, `js` (both Legacy and IR),
```
Comment (ilya-g) [inline README.md:12]: Please check the build. JS legacy is not supported in Kotlin 1.9

**Review by ilya-g — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Nailik) [reply README.md:12]: @qurbonzoda could you update this?

Comment (qurbonzoda) [reply README.md]: The JS Legacy support is now dropped

File: core/build.gradle.kts [hunk 3]
```diff
@@ -71,49 +79,32 @@ kotlin {
...
-
         val jsTest by getting {
-            dependencies {
-                api("org.jetbrains.kotlin:kotlin-test-js")
-            }
         }
 
         val nativeMain by getting {
-            dependsOn(commonMain.get())
+            dependsOn(commonMain)
```
Comment (ilya-g) [inline core/build.gradle.kts:104]: Optional: you can update infra plugin to 0.4.0-dev-80 to get rid of this explicit dependsOn.

File: core/build.gradle.kts [hunk 4]
```diff
@@ -44,7 +52,7 @@ kotlin {
         }
     }
 
-    js(BOTH) {
+    js(IR) {
```
Comment (ilya-g) [inline core/build.gradle.kts:55]: Minor: can be just `js { }`, `IR` is apparently the default backend for JS now.

File: gradle.properties [hunk 5]
```diff
@@ -4,11 +4,4 @@ versionSuffix=SNAPSHOT
...
-kotlin.mpp.enableGranularSourceSetsMetadata=true
-kotlin.mpp.enableCompatibilityMetadataVariant=true
-
-kotlin.native.distribution.type=prebuilt
-
-# Workaround for Bintray treating .sha512 files as artifacts
-# https://github.com/gradle/gradle/issues/11412
-systemProp.org.gradle.internal.publish.checksums.insecure=true
\ No newline at end of file
+kotlin.native.distribution.type=prebuilt
```
Comment (ilya-g) [inline gradle.properties:7]: Probably can be removed too, `prebuilt` is the default for a long time now.

**Review by ilya-g — APPROVED** [review]

File: core/build.gradle.kts (hunk 3 above)
Comment (qurbonzoda) [reply core/build.gradle.kts]: Done

**Comment by Zego001** [comment]
8c1a56755c579066071afb7a7f2e4c66cbb224b6

**Pull request merged**